### PR TITLE
Check file existence state before opening it  in cloud_cfg datasource

### LINKED ISSUE #2920: Unexpected Exception is thrown while running insights-client
```
# insights-client 
Starting to collect Insights data for slutade-rhel7.syslab.pnq2.redhat.com
Traceback (most recent call last):
  File "/var/lib/insights/last_stable.egg/insights/core/dr.py", line 962, in run
    result = DELEGATES[component].process(broker)
  File "/var/lib/insights/last_stable.egg/insights/core/dr.py", line 681, in process
    return self.invoke(broker)
  File "/var/lib/insights/last_stable.egg/insights/core/plugins.py", line 90, in invoke
    return self.component(broker)
  File "/var/lib/insights/last_stable.egg/insights/specs/default.py", line 282, in cloud_cfg
    with open(relative_path, 'r') as f:
IOError: [Errno 2] No such file or directory: '/etc/cloud/cloud.cfg'

```

The current `cloud_cfg` datasource caught the OSError instead of IOError.
https://github.com/RedHatInsights/insights-core/blob/1b2d8055dae1f6608901a1cc30f8ac978af71063/insights/specs/default.py#L288

### PR DESCRIPTION
**Pull request by xiangce**
Signed-off-by: Xiangce Liu <xiangceliu@redhat.com>

### CODE REVIEW

**Review by psachin — APPROVED** [review]
Tested locally by building lasted client egg.

File: insights/specs/default.py [hunk 1]
```diff
@@ -285,7 +285,7 @@ def cloud_cfg(broker):
                 if network_config:
                     return DatasourceProvider(content=json.dumps(network_config), relative_path=relative_path)
 
-        except OSError:
+        except IOError:
```
Comment (bfahr) [inline insights/specs/default.py:288]: @xiangce I believe this should be a generic `Exception` instead because we don't want this to raise any exceptions other than `SkipComponent` during collection.  

Also see my concerns about the original parser noted in the Issue #2920.  That should probably be done in another PR since any rules using this parser would need to change, but I'd prefer not to close the issue until that is fixed.

**Review by bfahr — CHANGES_REQUESTED** [review]

File: insights/specs/default.py (hunk 1 above)
Comment (xiangce) [reply insights/specs/default.py]: @bfahr - I add a safety check before opening the file, please check it again.

And I'm very sorry, I cannot get your concerns, would you please make it a bit clear?

Comment (xiangce) [reply insights/specs/default.py]: I just removed the `fix #2920`, which will keep it open after merging this PR.

Comment (xiangce) [reply insights/specs/default.py]: Just updated the #2920, and It seems that more customers are complaining about such problems.

Comment (bfahr) [reply insights/specs/default.py]: @xiangce not sure which point you are asking for more detail on.  

This should be a bare `Exception` to catch whatever problems may occur with reading the file, not just an `IOError`.

I opened a new Issue #2935 with more detail about the context the datasource is not capturing.

Comment (bfahr) [reply insights/specs/default.py]: @xiangce I discussed with @csams and we came up with the changes in my commit.  Also had to fix some merge conflicts to get this merged this morning.

**Review by lhuett — APPROVED** [review]
Looks good.
